app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java

```diff
@@ -1253,24 +1260,12 @@ private void dispatchHomeScreen() {
                     }
                 }
             }
-
-            //Now we need to catch any resource or database upgrade flags and make sure that the application
-            //is ready to go.
-            else if (CommCareApplication._().getAppResourceState() != CommCareApplication.STATE_READY ||
```

Comment:
Comment below on appresourcestate, I think we may need to clarify some additional states.